pandas/core/generic.py

```diff
@@ -5755,6 +5736,11 @@ def astype(self, dtype, copy=True, errors='raise', **kwargs):
                                          **kwargs)
             return self._constructor(new_data).__finalize__(self)
 
+        if not results:
```

Comment:
What case hits this? I'm not immediately seeing it.

Reply:
empty frames :->